syntaxes/kotlin.tmLanguage.json

```diff
@@ -385,11 +439,52 @@
                 }
             }
         },
+        "constant-declaration": {
+            "match": "\\b(val)\\b\\s*(?<GROUP><([^<>]|\\g<GROUP>)+>)?",
+            "captures": {
+                "1": {
+                    "name": "storage.type.variable.readonly.kotlin"
+                },
+                "2": {
+                    "patterns": [
+                        {
+                            "include": "#type-parameter"
+                        }
+                    ]
+                }
+            }
+        },
+        "variable": {
+            "match": "\\b(\\w+)(?=\\s*[:=])",
```

Comment:
This isn't quite right either, this will match anything on the left-hand side of `==` too, including e.g. numbers:

<img width="519" alt="image" src="https://github.com/fwcd/vscode-kotlin/assets/30873659/08e707ea-771e-4298-bf50-ed33545a75dd">

Anything involving `=` will likely require being very careful to not accidentally match any equality operators. This includes cases where e.g. the space is left out, i.e. expressions like `(x!=3)`...

Reply:
I'm removing this definition completely. It need a bit more work to be implemented correctly.